docs/src/main/asciidoc/qute-reference.adoc

```diff
@@ -489,29 +489,40 @@ This could be useful to simplify the template structure:
 <1> The `name` will be resolved against the `item.parent`.
 <2> The `description` will be also resolved against the `item.parent`.
 
-It's also possible to specify an alias for the context object:
+This section might also come in handy when we'd like to avoid multiple expensive invocations:
 
 [source,html]
 ----
-{#with item.parent as myParent} <1>
-  <h1>{myParent.name}</h1>
+{#with item.callExpensiveLogicToGetTheValue(1,'foo',bazinga)}
+  {#if this is "fun"} <1>
+    <h1>Yay!</h1>
+  {#else}
+    <h1>{this} is not fun at all!</h1>
+  {/if}
 {/with}
 ----
-<1> `myParent` is the alias that can be used inside the tag.
+<1> `this` is the result of `item.callExpensiveLogicToGetTheValue(1,'foo',bazinga)`. The method is only invoked once even though the result may be used in multiple expressions.
 
-This section might also come in handy when we'd like to avoid multiple expensive invocations:
+===== Let/Set Section
 
+This section is an alternative to `{#with}` which allows to specify one or more named parameters:
```

Comment:
No it is not: it does not change the scope: it adds new lexical bindings.

```suggestion
This section allows you to define named local variables:
```

Reply:
> No it is not: it does not change the scope: it adds new lexical bindings.

Not really. Under the hood it does modify the current context. But I can change the wording. No problem.

Reply:
OK, fine, it does introduce a new scope, right? It doesn't modify the current scope in the sense that it modifies the existing scope, right?

I still think that in this context, saying something along the lines of "it defines new local variables" is the easiest thing to explain to devs, no? We just want to make it clear that it's a bounded local scope with new bindings.